artichoke-backend/src/extn/core/encoding/trampoline.rs

```diff
@@ -0,0 +1,116 @@
+use super::Encoding;
+
+use crate::extn::core::array::Array;
+use crate::extn::core::string::{Encoding as SpinosoEncoding, String};
+
+use crate::extn::prelude::*;
+
+pub(super) const AVAILABLE_ENCODINGS: [Encoding; 3] = [Encoding::Utf8, Encoding::Ascii, Encoding::Binary];
+
+pub fn aliases(interp: &mut Artichoke) -> Result<Value, Error> {
+    let _ = interp;
+    Err(NotImplementedError::new().into())
+}
+
+pub fn compatible(interp: &mut Artichoke, lhs: Value, rhs: Value) -> Result<Value, Error> {
+    let _ = interp;
+    let _ = lhs;
+    let _ = rhs;
+    Err(NotImplementedError::new().into())
+}
+
+pub fn find(interp: &mut Artichoke, encoding: Value) -> Result<Value, Error> {
+    let _ = interp;
+    let _ = encoding;
+    Err(NotImplementedError::new().into())
+}
+
+pub fn list(interp: &mut Artichoke) -> Result<Value, Error> {
+    let _ = interp.encodings();
+    let _ = interp;
+    Err(NotImplementedError::new().into())
+}
+
+pub fn locale_charmap(interp: &mut Artichoke) -> Result<Value, Error> {
+    let _ = interp;
+    Err(NotImplementedError::new().into())
+}
+
+pub fn name_list(interp: &mut Artichoke) -> Result<Value, Error> {
+    let _ = interp;
+    Err(NotImplementedError::new().into())
+}
+
+#[allow(clippy::unnecessary_wraps)]
```

Comment:
this allow appears to be unused

Reply:
Yeah, heavy WIP. If it looks dirty, it'll be cleaned prior to requesting real review :smile: